src/modules/hip/kernel/normalize.hpp

```diff
@@ -0,0 +1,2004 @@
+        }
+        // reduce on XYZ block partial sums
+        else if(axisMask == 7)
+        {
+            partialSumBlocksPerSample = globalThreads_x * globalThreads_y;
+            hipLaunchKernelGGL(reduce_final_result_hip,
+                               dim3(ceil((float)partialSumBlocksPerSample/16), 1, srcGenericDescPtr->dims[0]),
+                               dim3(16, 1, 1),
+                               0,
+                               handle.GetStream(),
+                               partialSumArr,
+                               partialSumBlocksPerSample,
+                               meanTensor,
+                               stdDevTensor,
+                               isMean,
+                               roiTensor,
+                               axisMask,
+                               numDims);
+        }
+    }
+    else if(numDims > 3)
+    {
+        // interpret the input as 1D tensor
+        globalThreads_x = srcGenericDescPtr->strides[0];
+        globalThreads_y = 1;
+        globalThreads_z = srcGenericDescPtr->dims[0];
+        Rpp32u batchSize = globalThreads_z;
+
+        // allocate tensor for src strides
+        Rpp32u *srcMaxDims = handle.GetInitHandle()->mem.mgpu.scratchBuf.uintmem + (2 * batchSize * numDims);
+        Rpp32u *srcStrides = handle.GetInitHandle()->mem.mgpu.scratchBuf.uintmem + (3 * batchSize * numDims);
+        memcpy(srcMaxDims, &srcGenericDescPtr->dims[1], numDims * sizeof(Rpp32u));
+        memcpy(srcStrides, &srcGenericDescPtr->strides[1], numDims * sizeof(Rpp32u));
+
+        Rpp32u shared_memory_size = 0;
+        Rpp32u block_size = 1024;
+        if(maxParamVolume <= MAX_SHARED_MEMORY_SIZE)
+        {
+            if(maxParamVolume <= 32)
+                shared_memory_size = 32;
+            else if(maxParamVolume <= 64)
+                shared_memory_size = 64;
+            else if(maxParamVolume <= 128)
+                shared_memory_size = 128;
+            else if(maxParamVolume <= 256)
+                shared_memory_size = 256;
+            else if(maxParamVolume <= 512)
+                shared_memory_size = 512;
+            else
+                shared_memory_size = MAX_SHARED_MEMORY_SIZE;
+            block_size = shared_memory_size;
+        }
+
+        if(isMean)
+        {
+            hipLaunchKernelGGL(compute_mean_nd_hip_tensor,
+                               dim3(ceil((float)globalThreads_x/block_size), ceil((float)globalThreads_y), ceil((float)globalThreads_z)),
+                               dim3(block_size, 1, 1),
+                               shared_memory_size,
+                               handle.GetStream(),
+                               srcPtr,
+                               srcMaxDims,
+                               srcStrides,
+                               meanTensor,
+                               roiTensor,
+                               paramShape,
+                               paramStrides,
```

Comment:
@sampath1117 I think there are still too many variables being passed in at kernel launch. Please see if you can consolidate to passing a float2 for say srcDimsStrides_f2 or even better an _f4

Reply:
Combined multiple params into single param wherever possible
most of the params are arrays here like srcDims. srcStrides since it is a generic kernel